ignite/distributed/comp_models/base.py

```diff
@@ -216,10 +216,10 @@ def all_reduce(
         return cast(Union[torch.Tensor, float], self._collective_op(tensor, self._do_all_reduce, op, group=group))
 
     def all_gather(
-        self, tensor: Union[torch.Tensor, float, str], group: Optional[Any] = None
+        self, tensor: Union[torch.Tensor, float, str, Any], group: Optional[Any] = None
```

Comment:
Wouldn't you replace `float` with `Number` here?

Reply:
We had problems with Number previous as mypy and Number are not doing well. This may be changed since that